### DISCUSSION TITLE
Ошибка компиляции: exit status 1 Ошибка компиляции для платы NodeMCU 1.0 (ESP-12E Module).

### DISCUSSION THREAD

**Discussion by rodger-1** [vvip-68/GyverPanelWiFi#314] [Задавайте вопросы] [answered]
ArduinoIDE - 1.8.19
Версия ядра ESP8266 - 3.1.2   
Версия FastLED - 3.7.0
При компиляции выдает ошибку:
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z9breakTimexR12tmElements_t':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:151: multiple definition of `_Z9breakTimexR12tmElements_t'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:151: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z12refreshCachex':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:42: multiple definition of `_Z12refreshCachex'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:42: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z4yearx':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:137: multiple definition of `_Z4yearx'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:137: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z5monthx':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:128: multiple definition of `_Z5monthx'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:128: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z7weekdayx':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:119: multiple definition of `_Z7weekdayx'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:119: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z3dayx':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:110: multiple definition of `_Z3dayx'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:110: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z6secondx':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:101: multiple definition of `_Z6secondx'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:101: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z6minutex':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:92: multiple definition of `_Z6minutex'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:92: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z4isPMx':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:84: multiple definition of `_Z4isPMx'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:84: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z4isAMx':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:76: multiple definition of `_Z4isAMx'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:76: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z12hourFormat12x':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:62: multiple definition of `_Z12hourFormat12x'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:62: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z4hourx':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:53: multiple definition of `_Z4hourx'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:53: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z8makeTimeRK12tmElements_t':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:204: multiple definition of `_Z8makeTimeRK12tmElements_t'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:204: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z7setTimex':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:274: multiple definition of `_Z7setTimex'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:274: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:242: multiple definition of `getTimePtr'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:242: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z3nowv':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:250: multiple definition of `_Z3nowv'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:250: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z4yearv':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:133: multiple definition of `_Z4yearv'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:133: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z5monthv':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:124: multiple definition of `_Z5monthv'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:124: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z7weekdayv':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:115: multiple definition of `_Z7weekdayv'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:115: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z3dayv':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:106: multiple definition of `_Z3dayv'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:106: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z6secondv':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:97: multiple definition of `_Z6secondv'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:97: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z6minutev':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:88: multiple definition of `_Z6minutev'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:88: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z4isPMv':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:80: multiple definition of `_Z4isPMv'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:80: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z4isAMv':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:72: multiple definition of `_Z4isAMv'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:72: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z12hourFormat12v':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:58: multiple definition of `_Z12hourFormat12v'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:58: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z4hourv':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:49: multiple definition of `_Z4hourv'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:49: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z7setTimeiiiiii':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:286: multiple definition of `_Z7setTimeiiiiii'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:286: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z10adjustTimel':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:303: multiple definition of `_Z10adjustTimel'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:303: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z10timeStatusv':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:307: multiple definition of `_Z10timeStatusv'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:307: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z15setSyncProviderPFxvE':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:313: multiple definition of `_Z15setSyncProviderPFxvE'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:313: first defined here
c:/users/������/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: libraries\Time\TimeLib.cpp.o: in function `_Z15setSyncIntervalx':
C:\Users\������\Documents\Arduino\libraries\Time/TimeLib.cpp:319: multiple definition of `_Z15setSyncIntervalx'; libraries\Time\Time.cpp.o:C:\Users\������\Documents\Arduino\libraries\Time/Time.cpp:319: first defined here

**collect2.exe: error: ld returned 1 exit status
exit status 1
Ошибка компиляции для платы NodeMCU 1.0 (ESP-12E Module).**

**Comment by rodger-1** [comment]
Прошу помощи в решении проблемы

**Reply by vvip-68** [reply, marked as answer]
Я бы для начала удалил все библиотеки из папки Documents и перенес их в папку C:\Program Files (x86)\Arduino\libraries или собирал бы под пользователем у которого нет кириллицы в имени. Возможно версия Arduino IDE 1.8.19 не приемлет кириллицу в путях (вот те черные квадратики в логе). Не факт, но вдруг. Ну или для эксперимента поставил Arduino IDE 2.3.3 - возможно в ней нет проблемы с кириллицей в путях.

Если ии далее будут ошибки - нужно искать другие возможные причины.

**Reply by vvip-68** [reply]
И - а библиотеки точно из проекта взяты? Или какие-нибудь случайные?

**Reply by vvip-68** [reply]
А если из папки с библиотеками удалить папку Time - что-то поменяется?

**Reply by rodger-1** [reply]
Удалил все библиотеки, установил заново и все скомпилилось. Спасибо!

**Comment by phitrowkelvin** [comment]
если никто не против, у меня сейчас тож была проблема "Compilation error: exit status 1"
решилось после правки файла WString.cpp

size_t const String::FLT_MAX_DECIMAL_PLACES = 6;
size_t const String::DBL_MAX_DECIMAL_PLACES = 15;

и String::String(float value, ....
строка static size_t const DOUBLE_BUF_SIZE = (DBL_MAX_10_EXP + 1)  .... // dynamic
на 
static const size_t DOUBLE_BUF_SIZE = 128; // static

**Reply by vvip-68** [reply]
Я против. Вы внесли изменения в системную библиотеку ядра. И хорошо, если вы понимаете что делаете. А если нет - Бог знает в каких местах это вылезет... Например, судя по коду, в конструкторе строки вы изменили максимально допустимый размер строки на фиксированный в 128 байт (или символов?). А если строка будет длиннее? Значит она либо обрежется, либо не влезет в буфер, что вызовет крах приложения. Скомпилилось - хорошо. А вот на что эти изменения повлияют?
Но, может быть, я и ошибаюсь. Я не лез глубоко в код ядра / библиотеки.
